### PR TITLE
[AIRFLOW-3106] Validate Postgres connection after saving it

### PR DESCRIPTION
**Pull request by BasPH**
I want to create connection validation when saving a connection. I started with Postgres. Would be nice to know if you agree with the PR, before I implement validation for other connection types.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - [https://issues.apache.org/jira/browse/AIRFLOW-3106](https://issues.apache.org/jira/browse/AIRFLOW-3106)

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR adds validation for the Postgres connection type. After saving a Connection, it validates by creating a new database connection and executing `SELECT 1`. In case this fails, an error is displayed:

image

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/hooks/postgres_hook.py [hunk 1]
```diff
@@ -60,6 +60,15 @@ def get_conn(self):
         self.conn = psycopg2.connect(**conn_args)
         return self.conn
 
+    def validate_conn(self):
+        try:
+            conn = self.get_conn()
```
Comment (Fokko) [inline airflow/hooks/postgres_hook.py:65]: Can you change this to `with self.get_conn()`? So it closes automagically.

**Review by Fokko — COMMENTED** [review]
Awesome suggestion @BasPH 

I really like this and makes setting up connections through the UI much easier. 2 points:
- Can we also add this to the model hook? So we can generalize it also to other hooks than the Postgres one.
- Can you add it to the rbac ui as well? The old UI will be removed soon (I hope).

**Comment by ron819** [comment]
Isn't this PR a private case of this?
https://issues.apache.org/jira/projects/AIRFLOW/issues/AIRFLOW-2998

**Comment by ashb** [comment]
Yup, duplicate Jira issues. One should be closed as a duplicate of the other.

**Comment by ron819** [comment]
@ashb  not sure. This PR offers solution only for PostgreSQL.

**Comment by ashb** [comment]
Oh yes, I was basing my comment on the title of the this Jira - I've changed that to be more specific and linked the two issues.

**Comment by BasPH** [comment]
Thanks. Will see if I can process @Fokko 's comments tonight.

**Pull request closed without merging**